### PR TITLE
Allow sending with no payload, plus some small fixes

### LINKED ISSUE #10: sendWebPush() support no payload?
At the moment, the sendWebPush() method expects a payload and throws and error if there is no payload.

There are a few scenarios where this may not be desirable:

1.) Using existing subscription objects that dont have keys
2.) Desire to use tickles and not send payload for some messages
3.) Make it easier to drop in the library

@wibblymat thoughts on this?

Looking briefly through code, it would be a case of just checking input data and picking either sendWebPushTickle() sendWebPushWithData() OR single method (same as now), but switch between which headers are set.

@petele @paullewis @owencm thoughts on this?

### PR DESCRIPTION
**Pull request by wibblymat**
@gauntface 

Fixes #10

### CODE REVIEW

File: src/push.js [hunk 1]
```diff
@@ -35,26 +35,31 @@ function ub64(buffer) {
...
+ * @param  {String}   authToken     Optional token to be used in the
+ *                                  `Authentication` header if the endpoint
+ *                                  requires it.
+ * @param {Number}    paddingLength The number of bytes of padding to add to the
+ *                                  message before encrypting it.
  * @return {Promise} A promise that resolves if the push was sent successfully
  *                   with status and body.
  */
-function sendWebPush(message, subscription, authToken) {
+function sendWebPush(message, subscription, authToken, paddingLength) {
```
Comment (gauntface) [inline src/push.js]: Could you switch around the paddingLength with the authToken? When GCM goes away, the only change will be remove the authtoken.

I'm still pro just making a special method of GCM Auth header, simplifying sendWebPush method.

This change will also likely result in developers always sending the GCM API Key to all push services :(

**Comment by gauntface** [comment]
cc @petele 

Awesome - thanks for doing this.

Only concern is still with the auth token, I can see people doing:

```
subscriptions.forEach(subscription => {
    if (!subscription.aesgcm) {
        return;
    }
    pushEncryption.sendWebPush('Totes rad msg', subscription, GCM_API_KEY);
});
```

Which means they are sharing their API key with the world :(

If we changed back to a method, we can be smarter and only include the header for GCM requests.

```
pushEncryption.setGCMAPIKey(GCM_API_KEY);
subscriptions.forEach(subscription => {
    if (!subscription.aesgcm) {
        return;
    }
    pushEncryption.sendWebPush('Totes rad msg', subscription);
});
```

**Comment by addyosmani** [comment]
> If we changed back to a method, we can be smarter and only include the header for GCM requests.

I would be strongly in favour of this if we can push for it. I had similar concerns to @gauntface about API keys being shared with the world and can see folks tripping up here :/

**Comment by wibblymat** [comment]
@gauntface OK, done

File: src/push.js [hunk 2]
```diff
@@ -58,23 +67,31 @@ function sendWebPush(message, subscription, authToken) {
...
   // future.
   const endpoint = subscription.endpoint.replace(GCM_URL, TEMP_GCM_URL);
-
-  const payload = encrypt(message, subscription);
   const headers = {
-    'Content-Encoding': 'aesgcm',
-    'Encryption': `salt=${ub64(payload.salt)}`,
-    'Crypto-Key': `dh=${ub64(payload.serverPublicKey)}`
+    // TODO: Make TTL variable
+    Ttl: '0'
```
Comment (gauntface) [inline src/push.js]: Mega nit, shouldn't this be TTL similar to: https://github.com/gauntface/simple-push-demo/pull/28/files

**Comment by gauntface** [comment]
One small nit on TTL, other than that - LGTM

**Comment by wibblymat** [comment]
@gauntface you're right, spec says `TTL`. No idea why I thought otherwise!

**Pull request merged**